persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -77,29 +84,61 @@
 import org.slf4j.LoggerFactory;
 
 public class CassandraPersistence
-    implements Persistence<
-        Config, org.apache.cassandra.service.ClientState, org.apache.cassandra.service.QueryState> {
+    extends AbstractCassandraPersistence<
+        Config,
+        org.apache.cassandra.service.ClientState,
+        org.apache.cassandra.service.QueryState,
+        KeyspaceMetadata,
+        CFMetaData,
+        ColumnDefinition,
+        UserType,
+        IndexMetadata,
+        ViewDefinition> {
   private static final Logger logger = LoggerFactory.getLogger(CassandraPersistence.class);
 
-  private DataStore root;
   private CassandraDaemon daemon;
   private Authenticator authenticator;
   private QueryHandler handler;
   private QueryInterceptor interceptor;
 
+  // C* listener that ensures that our Stargate schema remains up-to-date with the internal C* one.
+  private MigrationListener migrationListener;
```

Comment:
Not strictly related to this PR, but what's the concurrency policy for this class? It initializes those non-final, non-volatile fields outside of the constructor, and CassandraPersistenceActivator doesn't use any synchronization when it calls `initialize`.

We should at least make the fields volatile, that will work if there won't be concurrent calls to `initializePersistence` (and I think that's a fair assumption).